### ISSUE TITLE
Still issue with Migration table in different schema (user) that currently is logged in

### ISSUE THREAD

**Issue by vasicvuk** [oracle/dotnet-db-samples#45]
I found here that the issue with different schema of migration table and logged user is solved but with latest version of NuGet package i still get following error:

```
 Oracle.ManagedDataAccess.Client.OracleException (0x80004005): ORA-00904: "T"."OWNER": invalid identifier
```

**Comment by alexkeh** [comment]
This is bug 30352492, which was just fixed. The fix missed the cutoff for the upcoming Oracle EF Core 19.5 release that will be on the NuGet Gallery soon. You can contact Oracle Support for a one-off fix or wait until the next NuGet release.

**Comment by vasicvuk** [comment]
When next NuGet release is planned? Currently i made a workaround by prefixing each Migration history table with Schema.

**Comment by alexkeh** [comment]
We haven't determined when to release again on NuGet Gallery after 19.5 yet.

It will depend on the timing of the Oracle EF Core 3 release and Oracle Developer Tools for VS 2019 coming out.

**Comment by vasicvuk** [comment]
But will EF Core 2.2 still be supported (and .Net Core 2.2) in the next release? Or maybe the question is if the current version will be patched at all?

Migration from EF Core 2.2 to 3.0 (and .Net Core 3) is a huge process and should not be done only to get this fix.

**Comment by alexkeh** [comment]
At some point, we will upload the latest Oracle EF Core 19.x patch for .NET Core 2.x to the NuGet Gallery. It's not clear when.

**Comment by vasicvuk** [comment]
Maybe the question is on another scope (I will post new issue if it is). 

Should EnsureSchema create User by it self. This currently works for other EF Core providers. Is this planned to be implemented?

**Comment by alexkeh** [comment]
EnsureCreated is not planned to create a user.

https://docs.oracle.com/en/database/oracle/oracle-data-access-components/19.3/odpnt/EFCoreDatabaseFacade.html#GUID-386D7462-8A37-4763-A009-01D92ED31A8A

The reason is if a user/schema is auto-created, what privileges to assign that user/schema by default? What if the EF Core user does not have privileges to assign those specific privileges to that user/schema?

**Comment by vasicvuk** [comment]
You are right but maybe this should be optional. In our usage case we have Microsevice architecture with 40 of microservices all using different schema (user in oracle case). Manual creation of all users is something that no one will like to do in this case. In this case it is easier to have user that can provision schemas by it self. 

Maybe adding some optional configuration in UseOracle options will satisfy both cases.

**Comment by alexkeh** [comment]
I can see the benefit of the feature, but I don't see a lot of demand for it compared to the work necessary. 

To make this feature robust would make it very complicated as there are A LOT of different privileges and roles an Oracle user/schema could be assigned. There's also tablespaces and sizes that need to be assigned.

I would recommend that if you are creating the same microservice user with the same privileges every time, then script those commands. Before calling EnsureCreated, execute the script. No manual steps needed from there,

**Comment by vasicvuk** [comment]
We are already doing that this way. But the official code is always better :)

**Comment by vasicvuk** [comment]
Anyway I am closing issue. We will contact Oracle directly for Patch if this is possible tommorow :)

**Comment by vasicvuk** [comment]
Hi @alexkeh,

Is this fixed in  2.19.50?

Thanks

**Comment by alexkeh** [comment]
No, it's not fixe din 2.19.50. A one-off patch has been approved and will be available in My Oracle Support. Your DBA should have access to this site. You will be able to download it there first. 

The fix will be in the next NuGet EF Core release, but we have not yet determined when we will release the next version yet.

**Comment by vasicvuk** [comment]
I looked at bug 30352492 on My Oracle support but no patches are available. Is this under some different bug ? 

Also, the guy from support tell me that he does not have any information from developers about the patch or if it was fixed at all

**Comment by alexkeh** [comment]
Sorry, to be more precise, the patch is not available to download today. It will be available on My Oracle Support soon. The patch was approved for a one-off 2 weeks ago by development.

If you'd like the support analyst to confirm, they can look at the bug log itself. 

An one-off patch request (30419058) was filed two weeks ago once Dev approved. That's currently in process of being created.

**Comment by alexkeh** [comment]
The fix for bug 30352492 is available in ODP.NET EF Core 19.6, which was just published on NuGet Gallery.
